### ISSUE TITLE
Trouble using events to switch pages

### ISSUE THREAD

**Issue by AalianKhan** [resoai/TileBoard#276]
I am having trouble switching pages using the events mechanic. I put this in my event tab in the developer section
![image](https://user-images.githubusercontent.com/39784853/67169328-d237ad00-f378-11e9-82fc-d578dd530a86.png)
But nothing happens on my iPad 2 or my desktop page. I even tried to implement it in automation but the same result. I don't even get an error. If someone can help me that will be great

**Comment by AalianKhan** [comment]
This might be related to a different problem I am having. Whenever I am refreshing the browser or logging in with my credentials, I always get an error at the bottom right corner saying unauthorized. Everything works when I toggle my lights and stuff but I get this error only when I refresh

**Comment by AalianKhan** [comment]
Also is there some sort of list for all the javascript commands

**Comment by AalianKhan** [comment]
If you mean the configuration file for tile board then yes I did. I copied all the events from Readme and paste it in the events section
`event: []`

**Comment by rchl** [comment]
1. Check you console in developer tools for any errors
2. Place a breakpoint in `function triggerEvents (eventData)` function and see if you are getting your tileboard event.

**Comment by AalianKhan** [comment]
Hey, thanks for the reply. I don't understand your second point. Where should I place `function triggerEvents (eventData)`
I try to look for a console in developer tools but couldn't find it. There were logs but when I fired the event nothing came up in the logs. I tried listening for the event tileboard in the event tab and I got the following.
![image](https://user-images.githubusercontent.com/39784853/67250755-9f55ed80-f43a-11e9-86de-ec1de1212315.png)

**Comment by AalianKhan** [comment]
@resoai Do you have any input in this?

**Comment by resoai** [comment]
You didn't show us what exactly you have in `CONFIG.events`.

**Comment by AalianKhan** [comment]
@resoai I said that I am using the same config provided in the readme file. Here is what I have currently but I am only using `open_page` command. 
```
   events: [
          /* Example: Start the screensaver on a tablet with Fully Kiosk Browser */
    {
      /* command: The command sent from Home Assistant */
      command: 'screen_off',

      /* action: Function to execute when the command is received
       * The variable e contains the full event_data from HomeAssistant
       */
      action: function(e) {
        if (typeof fully !== undefined) {
            fully.startScreensaver();
        }
      },
    },

    /* Example: End the screensaver and make sure Fully Kiosk Browser is in
     * the foreground.
     */
    {
      command: 'screen_on',
      action: function(e) {
        if (typeof fully !== undefined) {
          fully.stopScreensaver();
          fully.bringToForeground();
        }
      },
    },

    /* Example: Play a sound file
     * Include sound_url in the event_data from Home Assistant
     */
    {
      command: 'play_sound',
      action: function(e) {
        playSound(e.sound_url);
      }
    },

    /* Example: Open a specific TileBoard page
     * Include a page field in the event_data from Home Assistant
     * that matches the id: of a page in the TileBoard CONFIG
     */
    {
      command: 'open_page',
      action: function(e) {
        window.openPage(CONFIG.pages[e.page]);
      }
    }
   ],
```
This is a great resource and I wish to use it to its full potential. The worst thing about this project is its documentation and I will help you with that because that is the only thing I can help you with. I will start with the different types of tiles. and update you later.

**Comment by rchl** [comment]
> Hey, thanks for the reply. I don't understand your second point. Where should I place function triggerEvents (eventData)

I don't know how comfortable you are with browser developer tools but here are the steps:

1. Open TileBoard in your browser
2. Open developer tools (F12 or right click & inspect)
3. Find the code of the `function triggerEvents (eventData)` function. It's in main.js file around line 2000.
4. Place a breakpoint there and check if developer tools stop there when you trigger the event.

Can also check `Console` in developer tools to see if you get any errors reported.

I have tried triggering the event in my local setup and it worked.

**Comment by AalianKhan** [comment]
@rchl Hello, thanks for the reply
With some fiddling around I placed a breakpoint in `function triggerevents (eventdata)` 
![image](https://user-images.githubusercontent.com/39784853/68437539-4e126180-018f-11ea-93a0-f8977ecb828a.png)
But nothing happened on the tileboard and nothing came up in the console. 
However, whenever I refresh tileboard or load it up I get an `Unauthorized error`. when I go in the developer tools in Chrome I see this
![image](https://user-images.githubusercontent.com/39784853/68437707-b3fee900-018f-11ea-821f-c71fb0840f18.png)
Thank you for lending a hand 👍.

**Comment by AalianKhan** [comment]
Hello I fixed the unauthorized problem my adding a long-lived access token but now I keep getting the unknown error every few seconds
![image](https://user-images.githubusercontent.com/39784853/68440056-d7c62d00-0197-11ea-932c-9a4107561624.png)

**Comment by rchl** [comment]
I can't think of why this would be happening...

**Comment by AalianKhan** [comment]
> I can't think of why this would be happening...

Well I thank you for helping me learn basics of developer tools. Do you think it may be connected the new update where they switched from JSON to YAML.  I will try to use a unmodified version of tile board and let's see what will happen.

**Comment by AalianKhan** [comment]
I have installed a new instance in my Proxmox with a new tileboard setup but still, the problem persists. Now I am thinking that I am doing something wrong. This is what I am doing

I am putting this code block in my config.js file
```js
   events: [
      /* Example: Open a specific TileBoard page
      * Include a page field in the event_data from Home Assistant
      * that matches the id: of a page in the TileBoard CONFIG
      */
     {
       command: 'open_page',
       action: function(e) {
         window.openPage(CONFIG.pages[e.page]);
       }
     }
   ],
```
Then I refresh my tileboard page.
Now I go to the event section in dev tools and do the following
![image](https://user-images.githubusercontent.com/39784853/68910676-9e099f00-0720-11ea-85d4-3bb8e895c6c1.png)
What I think this event will do is move to the 2nd page of tileboard but when I trigger the event I see nothing. There are also no errors in the logs of the tileboard webpage. I also tried doing it on my phone but no luck there. I tried to make a tile that triggers the function `window.openPage` and it works. Please let me know if I am doing something wrong. thanks, @rchl @resoai 

Also, how is the documentation going? is there something I should change? I am going to start doing the different types of tiles there are and how to set them up. Basically modifying the tile example .md

**Comment by rchl** [comment]
Assuming that `events` array is in the correct place in your config.js (at the top level), I don't see anything wrong.
You can just put an `alert(1111)` line inside action handler like so: `action: function(e) { alert(111); }` to see for sure whether handler is triggered or not. If not, then the problem is probably with communication between tileboard and homeassistant. You could inspect `Network` tab in browser developer tools for  to check if the websocket connection is up and whether you are receiving any data on triggering event.

**Comment by AalianKhan** [comment]
Hey, thanks for the reply. I did the alert thing you wrote 
```js
     {
       command: 'open_page',
       action: function(e) {
         alert(111);
         window.openPage(CONFIG.pages[e.page]);
       }
     }
```
but I got nothing in the console. Then I checked my WebSocket and there is a connection there and in the messages, it is ping ponging (is that a word?) so I think that is not the problem either.
![image](https://user-images.githubusercontent.com/39784853/68963535-1d40b680-07a5-11ea-9427-0332a09e518e.png)
but I saw in the network tab that it is looking for a file named custom.css  at `http://MY-IP:8123/local/TileBoard-test/styles/custom.css?r=0.440960832112373` 
but it gives an error of 404 not found.
Also when I trigger the event and look in the messages tab in the WebSocket there is nothing there except for ping pongs
![image](https://user-images.githubusercontent.com/39784853/68964116-8d9c0780-07a6-11ea-9962-82f7ca056a15.png)


BTW this is all in the dev instance where nothing has been changed and I made it yesterday.
